src/v/kafka/client/direct_consumer/verifier/proto/verifier.proto

```diff
@@ -0,0 +1,156 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package redpanda.kafka.client;
+
+import "proto/redpanda/pbgen/options.proto";
+import "proto/redpanda/pbgen/rpc.proto";
+
+option (redpanda.pbgen.cpp_namespace) = "kafka::client::verifier";
+
+message BrokerAddress {
+    string host = 1;
+    int32 port = 2;
+}
+
+message DirectConsumerConfiguration {
+    int32 min_bytes = 1;
+    int32 max_fetch_size = 3;
+    int32 partition_max_bytes = 4;
+    enum OffsetResetPolicy {
+        EARLIEST = 0;
+        LATEST = 1;
+    }
+    OffsetResetPolicy reset_policy = 5;
+    int32 max_wait_time_ms = 6;
+    enum IsolationLevel {
+        READ_UNCOMMITTED = 0;
+        READ_COMMITTED = 1;
+    }
+    IsolationLevel isolation_level = 7;
+
+    // Queue settings
+    int64 max_buffered_bytes = 8;     // in bytes
+    int32 max_buffered_elements = 9;  // number of elements
+}
+
+message CreateDirectConsumerRequest {
+    string client_id = 1;
+    repeated BrokerAddress initial_brokers = 3;
+    DirectConsumerConfiguration consumer_configuration = 4;
+}
+
+message PartitionAssignment {
+    int32 partition_id = 1;     // Partition ID
+    optional int32 offset = 2;  // Offset to start consuming from
+}
+
+message TopicAssignment {
+    string topic = 1;
+    repeated PartitionAssignment partitions = 2;
+}
+
+message AssignPartitionsRequest {
+    string client_id = 1;
+    repeated TopicAssignment topic_assignments = 3;
```

Comment:
whats with these non con contiguous field tags? (just curious).

Reply:
should be 2 :)

Reply:
sorry, my mistake, fixing